ankaios_sdk/_components/workload_state.py

```diff
@@ -0,0 +1,397 @@
+        )
+
+
+class WorkloadStateCollection:
+    """
+    A class that represents a collection of workload states and provides
+    methods to manipulate them.
+    """
+    ExecutionsStatesForId: TypeAlias = dict[str, WorkloadExecutionState]
+    ExecutionsStatesOfWorkload: TypeAlias = dict[str, ExecutionsStatesForId]
+    WorkloadStatesMap: TypeAlias = dict[str, ExecutionsStatesOfWorkload]
+
+    def __init__(self) -> None:
+        """
+        Initializes a WorkloadStateCollection instance.
+        """
+        self._workload_states: list[WorkloadState] = []
+
+    def add_workload_state(self, state: WorkloadState) -> None:
+        """
+        Adds a workload state to the collection.
+
+        Args:
+            state (WorkloadState): The workload state to add.
+        """
+        self._workload_states.append(state)
+
+    def get_as_dict(self) -> WorkloadStatesMap:
+        """
+        Returns the workload states as a list.
+
+        Returns:
+            list[WorkloadState]: A list of workload states.
+        """
+        return_dict = self.WorkloadStatesMap()
+        for state in self._workload_states:
+
+            agent_name = state.workload_instance_name.agent_name
+            if agent_name not in return_dict:
+                return_dict[agent_name] = self.ExecutionsStatesOfWorkload()
+
+            workload_name = state.workload_instance_name.workload_name
+            if workload_name not in return_dict[agent_name]:
+                return_dict[agent_name][workload_name] = \
+                    self.ExecutionsStatesForId()
+
+            workload_id = state.workload_instance_name.workload_id
+            return_dict[agent_name][workload_name][workload_id] = \
+                state.execution_state
+        return return_dict
+
+    def get_as_list(self) -> list[WorkloadState]:
+        """
+        Returns the workload states as a list.
+
+        Returns:
+            list[WorkloadState]: A list of workload states.
+        """
+        return self._workload_states
+
+    def _from_proto(self, state: _ank_base.WorkloadStatesMap) -> None:
+        """
+        Populates the collection from a proto message.
+
+        Args:
+            state (_ank_base.WorkloadStatesMap): The proto message
+                to interpret.
+        """
+        for agent_name in state.agentStateMap:
+            for workload_name in state.agentStateMap[agent_name].\
+                    wlNameStateMap:
+                for workload_id in state.agentStateMap[agent_name].\
+                        wlNameStateMap[workload_name].idStateMap:
+                    self.add_workload_state(WorkloadState(
+                        agent_name,
+                        workload_name,
+                        workload_id,
+                        state.agentStateMap[agent_name]
+                        .wlNameStateMap[workload_name]
+                        .idStateMap[workload_id]
+                    ))
```

Comment:
The rationale was to have the collection as an unordered list of Workload States. Indeed I convert in `_from_proto` to a list, but it's from a proto message that I can't really use otherwise.

Also added a method to get the state of a workload, providing a workload instance name. This should improve the overall filtering.

Reply:
Now it should be fine.